### DISCUSSION TITLE
Should I wait for the broker.broadcast => Promise<void>?

### DISCUSSION THREAD

**Discussion by rulrok** [moleculerjs/moleculer#1067] [Q&A] [answered]
Hello,

I was wondering what is the difference if I do

`void broker.broadcast(...)`

and

`await broker.broadcast(...)`

The same question would be valid for `emit` method.

**Comment by shawnmcknight** [comment, marked as answer]
Personally, I don't see any reason to `await` the event response unless for some reason you don't wish to proceed with the rest of the execution flow until it returns.  In our project, we use `broker.broadcast(...).catch((err) => { ... log error ... });` so we can fire-and-forget the events but if there happens to be an error it gets logged.  We don't want it to throw in that case.

**Reply by rulrok** [reply]
I see. It uses the promise to leverage the error callback functionality then.
Should it just be a callback parameter then to avoid the doubt about the promise per se? I mean, since the promise result itself would not bring any value.
